test_scripts/API/Navigation/SendLocation/024_SendLocation_diff_HMI_levels.lua

```diff
@@ -0,0 +1,86 @@
+---------------------------------------------------------------------------------------------
+-- Requirements: https://github.com/smartdevicelink/sdl_requirements/blob/master/detailed_docs/TRS/embedded_navi/SendLocation_TRS.md
+--
+-- Requirement summary:
+-- 1. Request is sent in NONE, BACKGROUND, FULL, LIMITED levels
+-- 2. SDL responds DISALLOWED, success:false to request in NONE level and SUCCESS, success:true in other ones
+--
+-- Description:
+-- App requests SendLocation in different HMI levels
+--
+-- Steps:
+-- SDL receives SendLocation request in NONE, BACKGROUND, FULL, LIMITED
+--
+-- Expected:
+-- SDL responds DISALLOWED, success:false in NONE level, SUCCESS, success:true in BACKGROUND, FULL, LIMITED levels
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSendLocation = require('test_scripts/API/Navigation/commonSendLocation')
+
+--[[ General configuration parameters ]]
+config.application1.registerAppInterfaceParams.isMediaApplication = true
+config.application1.registerAppInterfaceParams.appHMIType = {"NAVIGATION"}
+config.application2.registerAppInterfaceParams.isMediaApplication = false
+config.application2.registerAppInterfaceParams.appHMIType = {"NAVIGATION"}
+
+--[[ Local Variables ]]
+local requestParams = {
+    longitudeDegrees = 1.1,
+    latitudeDegrees = 1.1
+}
+
+--[[ Local Functions ]]
+local function sendLocationDisallowed(params, self)
+    local cid = self.mobileSession2:SendRPC("SendLocation", params)
+
+    EXPECT_HMICALL("Navigation.SendLocation")
+    :Times(0)
+
+    self.mobileSession2:ExpectResponse(cid, { success = false, resultCode = "DISALLOWED" })
+    commonSendLocation.delayedExp(1000)
+end
+
+local function sendLocationSuccess(params, self)
+    local cid = self.mobileSession1:SendRPC("SendLocation", params)
+    params.appID = commonSendLocation.getHMIAppId()
+
+    EXPECT_HMICALL("Navigation.SendLocation", params)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function BringAppToLimitedLevel(self)
+	self.hmiConnection:SendNotification("BasicCommunication.OnAppDeactivated",
+		{ appID = self.applications["Test Application"] })
+
+	self.mobileSession1:ExpectNotification("OnHMIStatus", { hmiLevel = "LIMITED" })
+end
+
+local function BringAppToBackgroundLevel(self)
+	commonSendLocation.activateApp("2", self)
```

Comment:
@GetmanetsIrina It's possible to use int instead of string ("2")

Reply:
Done